msprime/ancestry.py

```diff
@@ -1698,7 +1705,38 @@ class DiracCoalescent(ParametricAncestryModel):
 @dataclasses.dataclass
 class SweepGenicSelection(ParametricAncestryModel):
     """
-    .. todo:: Document me
+    This simulates a coalescent history with a single selective
+    sweep that has occurred in the past. The model is one of a
+    a structured coalescent where selective backgrounds are defined as in
+    `Braverman et al. (1995) <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1206652/>`_
+    The implementation details here follow closely to those in discoal,
+    `Kern and Schrider (2016)
+    <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5167068/>`_
+
+    .. warning::
+        As the strength of selection (:math:`2Ns`) increases the chance
+        of two coalescent events happening very very close together in
+        time and can lead to msprime errors. should you encounter this
+        decrease :math:`2Ns` if appropriate.
+
+    :param float position: the location of the beneficial allele along the
+        chromosome.
+    :param float start_frequency: population frequency of the benefical
+        allele at the start of the selective sweep. e.g., for a *de novo*
+        allele in a diploid population of size N, start frequency would be
+        :math:`1/2N`. A soft selective sweep could be modelled by
+        setting start_frequency :math:`> 1/2N`.
+    :param float end_frequency: population frequency of the beneficial
+        allele at the end of the selective sweep. For a complete sweep
+        this should be set to :math:`(2N-1)/2N`. For a partial sweep,
+        this could be set to anything less than that value.
+    :param float alpha: alpha is the population scaled selection coefficient
+        :math:`2Ns`. we define s such that the fitness of the three genotypes
```

Comment:
"We define". Lots of these problems - documentation is real, written English!

Reply:
Maybe the selection coeffecient deserves its own paragraph above? In particular, I can see "genotypes" being confusing to some people, as this is contrary to how we use genotype in tskit.

Reply:
Can we write this in a way that makes sense if N changes over the course of the simulation? (eg. maybe alpha is, approximately, the rate of appearance of new selected-mutations-which-make-it-to-fixation?)

Reply:
moved this up and tried to make it a bit more clear

Reply:
> Can we write this in a way that makes sense if N changes over the course of the simulation? (eg. maybe alpha is, approximately, the rate of appearance of new selected-mutations-which-make-it-to-fixation?)

@petrelharp I think the next step is to generalize the implementation so that s is separate from N. this will take only take a bit of effort to rewrite the trajectory generation routine but will also allow us to allow N to change during the sweep. 

I'm hoping we can leave it like this for the initial release and then I will get the second, more general implementation in place.

Reply:
That's fine; just wanting to make clear what Ne means here. The docs do a good job; I'll comment there.